mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp

```diff
@@ -65,12 +65,12 @@ void ConcatOp::getCanonicalizationPatterns(RewritePatternSet &results,
 }
 
 LogicalResult SelectOp::canonicalize(SelectOp op, PatternRewriter &rewriter) {
-  auto notOp = op.getPred().getDefiningOp<tosa::LogicalNotOp>();
+  auto notOp = op.getInput1().getDefiningOp<tosa::LogicalNotOp>();
```

Comment:
Feel this change is unrelated to this patch. Can we pull into a separate one?

Reply:
That's right. Will put that in another patch.

Reply:
Separate patch here: https://github.com/llvm/llvm-project/pull/127833